tarsier/tests/test_elements.py

```diff
@@ -0,0 +1,130 @@
+import os
+import pytest
+from playwright.async_api import async_playwright
+from dotenv import load_dotenv
+from typing import Dict
+from tarsier import GoogleVisionOCRService, Tarsier
+
+dotenv_path = os.path.join(os.path.dirname(__file__), "test.env")
+load_dotenv(dotenv_path)
+
+
+def google_creds() -> Dict[str, str]:
+    return {
+        "type": os.getenv("TYPE") or "",
+        "project_id": os.getenv("PROJECT_ID") or "",
+        "private_key_id": os.getenv("PRIVATE_KEY_ID") or "",
+        "private_key": (os.getenv("PRIVATE_KEY") or "").replace("\\n", "\n"),
+        "client_email": os.getenv("CLIENT_EMAIL") or "",
+        "client_id": os.getenv("CLIENT_ID") or "",
+        "auth_uri": os.getenv("AUTH_URI") or "",
+        "token_uri": os.getenv("TOKEN_URI") or "",
+        "auth_provider_x509_cert_url": os.getenv("AUTH_PROVIDER_X509_CERT_URL") or "",
+        "client_x509_cert_url": os.getenv("CLIENT_X509_CERT_URL") or "",
+        "universe_domain": os.getenv("UNIVERSE_DOMAIN") or "",
+    }
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "html_file, expected_tag_to_xpath",
+    [
+        ("test_html/text_only.html", {0: "//html/body/h1", 1: "//html/body/p"}),
+        (
+            "test_html/hyperlink_only.html",
```

Comment:
Let's also have some expectation for page to text. (we can simplify the html to handle this and ensure there's only 1 /2 rows per test except the combination test